x-pack/plugins/search_indices/public/components/index_documents/add_documents_code_example.tsx

```diff
@@ -5,19 +5,155 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { MappingProperty } from '@elastic/elasticsearch/lib/api/types';
+import { EuiFlexGroup, EuiFlexItem, EuiPanel, EuiText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { TryInConsoleButton } from '@kbn/try-in-console';
 
-import { EuiPanel } from '@elastic/eui';
+import { useKibana } from '../../hooks/use_kibana';
+import { IngestCodeSnippetParameters } from '../../types';
+import { LanguageSelector } from '../shared/language_selector';
+import { useIngestCodeExamples } from './hooks/use_ingest_code_examples';
+import { useElasticsearchUrl } from '../../hooks/use_elasticsearch_url';
+import { useUsageTracker } from '../../contexts/usage_tracker_context';
+import { AvailableLanguages, LanguageOptions, Languages } from '../../code_examples';
+import { AnalyticsEvents } from '../../analytics/constants';
+import { CodeSample } from '../shared/code_sample';
+import { generateSampleDocument } from '../../utils/document_generation';
+
+export interface AddDocumentsCodeExampleProps {
+  indexName: string;
+  mappingProperties: Record<string, MappingProperty>;
+}
+
+export const AddDocumentsCodeExample = ({
+  indexName,
+  mappingProperties,
+}: AddDocumentsCodeExampleProps) => {
+  const { application, share, console: consolePlugin } = useKibana().services;
+  const ingestCodeExamples = useIngestCodeExamples();
+  const elasticsearchUrl = useElasticsearchUrl();
+  const usageTracker = useUsageTracker();
+  const indexHasMappings = Object.keys(mappingProperties).length > 0;
+
+  // TODO: initing this should be dynamic and possibly saved in the form state
+  const [selectedLanguage, setSelectedLanguage] = useState<AvailableLanguages>('python');
```

Comment:
I wonder if for now we store this option in localstorage and have a hook so the selected language is consistent everywhere. 

Ideally i would love a user preference for this that also part of the dev console code exporter too.

Reply:
I want to say the dev console is doing this, let me check where they are storing that and maybe we use that.

The bigger difference will be they have an explicit "set as default" option, so I'm not 100% sure if we'd want to update that value if we change it here. But if that value is set we could default to the same language?

Either way I want to write a utility function for find the default language and we can iterate  on the logic for that.